cishouseholds/impute.py

```diff
@@ -110,19 +147,76 @@ def fill_forward_from_last_change(
     minimum_dateset_version
         minimum dataset version that should be filled from
     """
+
+    df_fill_forwards_from = generate_fill_forward_df(
+        df,
+        fill_forward_columns,
+        participant_id_column,
+        visit_datetime_column,
+        record_changed_column,
+        record_changed_value,
+    )
+    df_filtered = df.filter(F.col(dateset_version_column) >= minimum_dateset_version)
+    df_filtered = fill_forward_from_last_change_process(
+        df_filtered, fill_forward_columns, participant_id_column, visit_datetime_column, df_fill_forwards_from
+    )
+    return df_filtered.union(df.drop("ROW_NUMBER").filter(F.col(dateset_version_column) < minimum_dateset_version))
```

Comment:
Do we need this `drop` here?

Also, can we refactor the filter condition to an object and use the inverse (`~condition`) here to ensure that they are exclusive?